### ISSUE TITLE
v2.0.1 : WGBS fails with name 'pipeline' is not defined because of FastQC.snakefile STARsolo changes

### ISSUE THREAD

**Issue by gtrichard** [maxplanck-ie/snakepipes#584]
This is the command I used with a newly installed 2.0.1 environment (I ran `createEnvs` and `config`)

```
$ WGBS -i /home/genouest/inra_umr1349/grichard/Data/BP/BS_REACTION/SnakePipes/input -o Chiifu_test/ -c configWGBS.yaml chiifu3


Sample sheet found and format is ok!

---- This analysis has been done using snakePipes version 2.0.1 ----
Sample sheet found and format is ok!
NameError in line 1 of /home/genouest/inra_umr1349/grichard/miniconda3/envs/snakePipes2/lib/python3.6/site-packages/snakePipes/shared/rules/FastQC.snakefile:
name 'pipeline' is not defined
  File "/home/genouest/inra_umr1349/grichard/miniconda3/envs/snakePipes2/lib/python3.6/site-packages/snakePipes/workflows/WGBS/Snakefile", line 37, in <module>
  File "/home/genouest/inra_umr1349/grichard/miniconda3/envs/snakePipes2/lib/python3.6/site-packages/snakePipes/shared/rules/FastQC.snakefile", line 1, in <module>
Error: snakemake returned an error code of 1, so processing is incomplete!
```

Replacing `FastQC.snakefile 2.0.1` with the `FastQC.snakefile 1.3.2` is fixing the issue. Hence the problem comes from the changes brought by 2.0.1, i.e. the STARsolo statements:

https://github.com/maxplanck-ie/snakepipes/blob/fd1d162d4fa633ea178d2f84ffac547cf17753e5/snakePipes/shared/rules/FastQC.snakefile#L1

Best,
Gautier

**Comment by katsikora** [comment]
Morning Gautier, thanks for spotting the issue!
So the actual problem arises as WGBS doesn't have the 'pipeline' keyword in its defaults.yaml. Every other workflow does. Can you fix this for yourself by putting:
pipeline: WGBS
there? I'll add the fix to develop in the meantime.

**Comment by katsikora** [comment]
I have to say I'm surprise this passed the dag test.

**Comment by gtrichard** [comment]
Yup it's really weird the dag test and CI passed, at first I thought I did something wrong installing the new environment.

**Comment by katsikora** [comment]
Gautier, did you set fastqc to True in your configWGBS.yaml?
I think this might be the difference between your cmd and our dag tests.

**Comment by gtrichard** [comment]
Yes it was set on true.

```
## General/Snakemake parameters, only used/set by wrapper or in Snakemake cmdl, but not in Snakefile
outdir:
configFile:
clusterConfigFile:
local: False
maxJobs: 12
## directory with fastq or bam files
indir:
## Genome information
genome:
###SNP black list (bed file)
blacklist:
###sample Sheet
sampleSheet: samples.txt
###inclusion bounds for methylation extraction
noAutoMethylationBias: False
## FASTQ file extension (default: ".fastq.gz")
ext: '.fastq.gz'
## paired-end read name extension (default: ['_R1', "_R2"])
reads: ['_R1', '_R2']
## Number of reads to downsample from each FASTQ file
downsample:
## Options for trimming
trim: True
trimmer: 'fastp'
trimmerOptions: '-q 5 -l 30 -M 5 --trim_front1 15 --trim_front2 15'
## Bin size of output files in bigWig format
bwBinSize: 25
## Run FASTQC read quality control
fastqc: True
verbose: False
plotFormat: 'pdf'
#### Flag to control the pipeline entry point
fromBAM: False
bamExt: '.bam'
pairedEnd: True
###Flags to control skipping of certain QC calculations
skipDOC: False
GCbias: False
###Thresholds for filtering of statistical comparisons (DMRs and DMLs)
DMRprograms: 'metilene,dmrseq'
maxDist: 300
minCpGs: 10
minCoverage: 5
FDR: 0.1
minMethDiff: 0.1
###MethylDackel options
MethylDackelOptions: '--mergeContext --maxVariantFrac 0.25 --minDepth 4 --CHH --CHG'
##umi_tools
UMIBarcode: False
bcPattern: NNNNCCCCCCCCC #default: 4 base umi barcode, 9 base cell barcode (eg. RELACS barcode)
UMIDedup: False
UMIDedupSep: "_"
UMIDedupOpts: --paired
aligner: bwameth

```

**Comment by katsikora** [comment]
Thanks!

**Comment by katsikora** [comment]
this is now fixed in develop PR#590
